contributors/devel/release/testing.md

```diff
@@ -0,0 +1,143 @@
+# Kubernetes test sustaining engineering
+
+This document describes how Kubernetes automated tests are maintained as part
+of the development process.
+
+## Definitions
+
+The following definitions are for tests continuously run as part of CI.
+
+- *test*
+  - *artifact*: a row in [test grid]
+  - a single test run as part of a test job
+  - maybe either an e2e test or an integration / unit test
+- *test job*
+  - *artifact*: a tab in [test grid]
+  - a collection of tests that are run together in shared environment. may:
+    - run in a specific environment - e.g. [gce, gke, aws], [cvm, gci]
+    - run under specific conditions - e.g. []upgrade, version skew, soak, serial]
+    - test a specific component - e.g. [federation, node]
+- *test infrastructure*
+  - not directly shown in the test grid
+  - libraries and infrastructure common across tests
+- *test failure*
+  - persistently failing test runs for a given test
+- *test flake*
+  - periodically failling test runs for a given test
+  
+## Ownership
+
+Each test must have an escalation point (email + slack).  The escalation point is responsible for
+keeping the test healthy.  Fixes for test failures caused by areas of ownership outside the
+responsibility of the escalation point should be coordinated with other teams by the
+test escalation point.
+
+Escalation points are expected to be responsive within 24 hours, and prioritize test failure
+issues over other issues.
+
+### test
```

Comment:
> I see the problem with the fact that in single suite (aka 'test-job') we're running tests owned by pretty much all SIGs. This is for legacy reasons.

I agree this isn't ideal, but it is what we have now.  I agree having ownership scoped to jobs would be ideal, but I don't want to block ownership of out existing tests on building something new.  I can add this as a suggestion for SIGs to move the tests they care about into separate jobs.

If we want to keep tests on master healthy for 1.8 (we do), we need to come up with a viable solution that we can execute in the next few weeks.